### PR TITLE
Feature/task24

### PR DESCRIPTION
**Pull request by 443n0511**
# [Issue No.24](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#24)
- 会員登録画面の作成
## [CodeSandBox](https://codesandbox.io/s/github/443n0511/js-lesson/tree/feature/task24/lesson24)<br>
Latest Commit: 582c8ac

## Concerns and areas to focus on
- This time, CSS was rewritten to tailwindCSS. I'm sure I'm missing something, but I hope you'll check it out.
- 9fd95a0

### CODE REVIEW

**Comment by nakagawamai** [comment]
IMO 
The label seems to cover the input area when not in focus. 
It would be more user friendly if they didn't cover the input area after typing.

https://user-images.githubusercontent.com/99260576/190935957-6a48e9fe-b40a-44fc-b7be-e3ccd519c58a.mov

**Comment by nakagawamai** [comment]
I think the whole process of getting the Agree button ticked on the Terms of Use is brilliant!
As a user, it is a stress-free UI/UX.

**Comment by csakyo** [comment]
> IMO The label seems to cover the input area when not in focus. It would be more user friendly if they didn't cover the input area after typing.
> 
>  2022-09-19.9.30.21.mov

I thought the same thing!

File: lesson24/app.js [hunk 1]
```diff
@@ -133,8 +128,8 @@ const setAgree = () => {
 
 const confirmationInput = (e) => {
   if (e.target.value === "" || !e.target.classList.contains("input"))
-    return e.target.nextElementSibling.classList.remove("enterd");
-  e.target.nextElementSibling.classList.add("enterd");
+    return e.target.nextElementSibling.classList.remove("");
+  e.target.nextElementSibling.classList.add("");
```
Comment (nakagawamai) [inline lesson24/app.js:132]: Q
removeするクラス名と、addするクラス名は空欄ですが、これは意図していますか？
下記のようなエラーが発生しているようです。
インプットエリアにラベルが覆い被さっている原因かもしれません。
スクリーンショット 2022-09-19 21 43 36

**Comment by nakagawamai** [comment]
I have a question about for the ` moveFocus` function and `handleKeydownModal` function. 
The following code.
Why should it be stored in an array and join?
````JavaScript 
    modalStatus.focusableElements = [
      ...modal.querySelectorAll(FOCUSABLE_ELEMENTS.join(","))
    ];
```` 

The following code works fine too.
````JavaScript 
    modalStatus.focusableElements = modal.querySelectorAll(FOCUSABLE_ELEMENTS); 
````

File: lesson24/css/style.css [hunk 2]
```diff
@@ -1,399 +1,56 @@
...
   }
 }
 
-@keyframes change-color__primary {
-  from {
-    background-color: var(--color-primary);
-    color: var(--color-white);
+@layer utilities {
+  .button__primary {
+    @apply bg-primary text-white focus:bg-white focus:text-primary focus-visible:outline-none focus-visible:border-primary focus-visible:border-2 hover:enabled:animate-change-color__primary hover:enabled:border-2 hover:enabled:border-primary;
```
[the remark this answers is not in the document]
Comment (443n0511) [reply lesson24/css/style.css]: I thought it was very difficult to read, so I was very impressed. Thanks for letting me know!
4b7acf1

File: lesson24/app.js (hunk 1 above)
Comment (443n0511) [reply lesson24/app.js]: Sorry for the incomplete PR.
Please confirm that it has been corrected.
576700b

**Comment by 443n0511** [comment]
@nakagawamai 
> I have a question about for the moveFocus function and handleKeydownModal function.
> The following code.
> Why should it be stored in an array and join?

Mai-san is right, we don't need join() in the code here.

In `handleKeydownModal`, I need to operate with an array instead of a nodeList, so we did the following.
```js
  modalStatus.focusableElements = [
    ...modal.querySelectorAll(FOCUSABLE_ELEMENTS),
  ];
```
582c8ac

**Review by csakyo — APPROVED** [review]

**Pull request merged**